src/hw/checkCurrencyApp.js

```diff
@@ -0,0 +1,33 @@
+// @flow
+import Transport from "@ledgerhq/hw-transport";
+import getAppAndVersion from "./getAppAndVersion";
+import {
+  DeviceOnDashboardUnexpected,
+  WrongAppForCurrency
+} from "@ledgerhq/errors";
+import type { CryptoCurrency } from "../types";
+
+export default async (
+  transport: Transport<*>,
+  currency: CryptoCurrency
+): Promise<void> => {
+  const currentApp = await getAppAndVersion(transport).catch(e => {
+    if (e.status === 0x6e00) return null;
+    throw e;
+  });
+
+  if (currentApp) {
+    if (currentApp.name === currency.managerAppName) {
+      return;
+    } else if (
+      // nano x 1.2.4-1 dashboard app name
+      ["BOLOS", "OLOS\u0000"].includes(currentApp.name)
+    ) {
+      throw new DeviceOnDashboardUnexpected();
+    }
+    throw new WrongAppForCurrency(
+      `wrong app ${currentApp.name}, expected ${currency.managerAppName}`,
+      { current: currentApp.name, expected: currency.managerAppName }
+    );
+  }
```

Comment:
as a else case, we should fallback on just checking a getAddress works


```
await getAddress(transport, {
  derivationName: "",
  devicePath,
  currency,
  path: runDerivation....(),
  segwit: false,
})
```

Reply:
:ah: